packages/steiger/src/features/remove-global-ignores-from-vfs/remove-global-ignores-from-vfs.ts

```diff
@@ -0,0 +1,17 @@
+import { Folder, GlobalIgnore } from '@steiger/types'
+import { applyExclusion } from '../../shared/globs'
+import { mergeGlobGroups } from '../../shared/globs/merge-glob-groups'
+
+function turnGlobalIgnoresIntoSingleGlobGroup(globalIgnores: Array<GlobalIgnore>) {
+  return mergeGlobGroups(globalIgnores.map(({ ignores }) => ({ files: [], ignores })))
```

Comment:
issue: I'm kind of concerned with this operation that adds `files: []` to an object from a semantics perspective. The way the config file is defined, you can do either:
- specify just `files`, which will only include the files you specified
- specify both `files` and `ignores`, which will include the files you specified except for the ones that are covered by ignores
- specify only `ignores`, which will include all the files except for the ones that are covered by ignores

Therefore, files and ignores can be optional, and we should support that on the exclusion operator level. `{ files: [], ignores: ['*.js'] }` is not the same as `{ ignores: ['*.js'] }` semantically. The first one selects nothing, the second one selects everything except for top-level JS files.

suggestion: let's remove this mapping and adjust the exclusion operator to differentiate between these two cases

Reply:
This was a kind of foolproof mechanism on my part (to sort out situations where users think that an empty array simply selects everything (when in files) or does nothing (when in ignores)), allowing empty files/ignores arrays in these objects to be treated as if they weren't there at all. Because I didn't see any useful applications of such arrays when they were empty. There is literally no case where a user wants to use them intentionally. But, ok, I can make it more strict if you think it should be. Or alternatively, I can handle cases when they are empty more gracefully (without the need to add empty file arrays before glob exclusion). I think we can allow empty arrays in raw configs but remove them during the config parsing stage, so all other modules are not infected with this knowledge. What do you think?

Reply:
I would rather keep the logic predictable and do any fool-proofing on the config validation stage by printing warnings about potential errors. Let's make it more strict for now and we can always improve on the UX later if it turns out to be a problem.

> I think we can allow empty arrays in raw configs but remove them during the config parsing stage

That would be quite unpredictable, so I don't think we should go with this approach

Reply:
Got it, I don't mind and have made the necessary fixes